frontend/src/api/queries/Market.ts

```diff
@@ -75,50 +80,88 @@ const marketFromResponse = (response: ResponseMarket): Market => {
     depositStopDate: new Nanoseconds(response.deposit_stop_date),
     withdrawalStopDate: new Nanoseconds(response.withdrawal_stop_date),
     winnerOutcome: typeof response.winner === "number"
-      ? outcomes.find(outcome => outcome.id === `${response.winner}`)
+      ? outcomes.find((outcome) => outcome.id === `${response.winner}`)
       : undefined,
     totalWallets: response.total_wallets,
     poolSize: BigNumber(response.pool_size),
+    lpShares: BigNumber(response.lp_shares),
   }
 }
 
-const outcomeFromResponse = (market: ResponseMarket, response: ResponseMarketOutcome): MarketOutcome => {
-  let totalTokens = BigNumber(0);
+const outcomeFromResponse = (
+  market: ResponseMarket,
+  response: ResponseMarketOutcome,
+): MarketOutcome => {
+  let totalPoolTokens = BigNumber(0)
   for (const outcome of market.outcomes) {
-    totalTokens = totalTokens.plus(outcome.total_tokens)
+    totalPoolTokens = totalPoolTokens.plus(outcome.pool_tokens)
+  }
+  const outcomeTotalTokens = BigNumber(response.total_tokens)
+  const outcomePoolTokens = BigNumber(response.pool_tokens)
+
+  // Taken from: https://docs.gnosis.io/conditionaltokens/docs/introduction3/
+  // oddsWeightForOutcome = product(numOutcomeTokensInInventoryForOtherOutcome for every otherOutcome)
+  // oddsForOutcome = oddsWeightForOutcome / sum(oddsWeightForOutcome for every outcome)
+
+  const oddsWeights = []
+  let totalOddsWeights = BigNumber(0)
+  let totalProduct = BigNumber(1)
+  for (let i = 0; i < market.outcomes.length; ++i) {
+    totalProduct = totalProduct.times(market.outcomes[i].pool_tokens)
+    let oddsWeight = BigNumber(1)
+    for (let j = 0; j < market.outcomes.length; ++j) {
+      if (i !== j) {
+        oddsWeight = oddsWeight.times(market.outcomes[j].pool_tokens)
+      }
+    }
+    oddsWeights.push(oddsWeight)
+    totalOddsWeights = totalOddsWeights.plus(oddsWeight)
   }
-  const outcomeTotalTokens = BigNumber(response.total_tokens);
+  const oddsForOutcome = totalProduct
+    .div(response.pool_tokens)
+    .div(totalOddsWeights)
+  let oddsWeightForOutcome = BigNumber(1)
+  for (const outcome of market.outcomes) {
+    oddsWeightForOutcome = oddsWeightForOutcome.times(outcome.pool_tokens)
+  }
+  oddsWeightForOutcome = oddsWeightForOutcome.div(response.pool_tokens)
```

Comment:
@lvn-rusty-dragon coming back to this I realized that `oddsWeightForOutcome` is not being used

Reply:
Doh, you're right, looks like I ended up implementing things twice. You mind cleaning that up? Sorry.

Reply:
So just not using `oddsWeightForOutcome` at all, right?

Reply:
Correct. Though probably hold off on this until the new math model is in place.